### ISSUE TITLE
Silently dropped acks and nacks during subscriber shutdown in LeaseLoop

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6675]
Unconfirmed.  This is the result of a search with Gemini, may be a false positive.

When shutting down the subscriber lease loop, `shutdown<L>` drains outstanding actions from the application's unbounded channel (`ack_rx`):

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/pubsub/src/subscriber/lease_loop.rs#L145-L155

Specifically lines 149–153:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/pubsub/src/subscriber/lease_loop.rs#L149-L153

The drain loop explicitly matches only `Action::Ack`:
```rust
while let Ok(r) = ack_rx.try_recv() {
    if let Action::Ack(ack_id) = r {
        state.process(Action::Ack(ack_id));
    }
}
```
Any `Action::Nack`, `Action::ExactlyOnceAck`, or `Action::ExactlyOnceNack` remaining in `ack_rx` are silently discarded.

As a result, messages negatively acknowledged before shutdown cannot be redelivered until server-side lease expiration, and callers awaiting exactly-once ack/nack confirmations fail or hang due to dropped oneshot response channels. Instead of filtering for `Action::Ack`, `state.process(r)` should be invoked for all drained actions.

**Comment by dbolduc** [comment]
Note that (1) we do not support waiting for exactly once acks when configured to NackImmediately https://github.com/googleapis/google-cloud-rust/issues/5109 and (2) all known messages are nacked, so it is not like we are dropping nacks.
